property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java

```diff
@@ -61,10 +59,38 @@ public void validateCreateRequest(PropertyRequest request) {
 
 		Map<String, String> errorMap = new HashMap<>();
 
+		validateRole(request, errorMap);
 		validateOwner(request, errorMap);
 
 	}
 
+	private void validateRole(PropertyRequest request, Map<String, String> errorMap) {
+		
+		//fetch all userinfo roles...
+		RequestInfo requestInfo = request.getRequestInfo();
+		List<org.egov.common.contract.request.Role> roleList = null;
+		if(null != requestInfo.getUserInfo()) {
+			roleList = requestInfo.getUserInfo().getRoles();
+			List<String> roleCode =  new ArrayList<String>(0);
+					
+			roleList.stream().forEach(r -> {
+				roleCode.add(r.getCode());
+			});
+		}
+		
+		
+		//check with proerpty have branch type as per userinfo role
+		List<Property> propertyList = request.getProperties();
+		for (Property property_ : propertyList) {
+			if(null != property_.getPropertyDetails() && 
+					!roleList.contains(property_.getPropertyDetails().getBranchType())){
```

Comment:
Role will be like `EB_APPROVER` but the branch type will be `EstateBranch`. Both will not be the same.

Reply:
Ok Agree but need to discussion ... currently in request we have branch type and role have EB_APPROVE ...s o maching base on string matches ?